modin/pandas/groupby.py

```diff
@@ -978,6 +935,63 @@ def _wrap_aggregation(
             return result.squeeze()
         return result
 
+    def _check_result_index(self, func):
+        """
+        Check the result of `func` on the need of resetting index.
+
+        Parameters
+        ----------
+        func : callable
+            The function, the result of which will be checked.
+
+        Returns
+        -------
+        DataFrame
+            The result of `func` with dropped index or an original result.
+        """
+
+        def wrapper(*args, **kwargs):
```

Comment:
This looks like over-engineering a bit, you can make `_check_result_index` accept the function and its arguments.
In the current usage the fact that it's a wrapper doesn't give you anything but complicates the code a little bit.

Or (what is probably better) you could make `wrapper()` accept also `self` and make two checking methods like this:
```python
    def _apply_agg_function(self, ...):
        # do the thing
    _check_reset_index_agg_func = _check_result_index(_apply_agg_function)
    _check_index_name_agg_func = _check_result_index_name(_apply_agg_function)
```

and then use the two wrapped functions instead of explicitly wrapping the `_apply_agg_function).

That said, I don't see this wrapper being used anywhere else (or with any other function), maybe just implement two more functions calling the `_apply_agg_function` directly (without implementing them as decorators)?

Reply:
Functions `_apply_agg_function_check_index` and `_apply_agg_function_check_index_name` have been added instead of previous `wrapper` functions.